tests/test_series.py

```diff
@@ -3285,9 +3286,45 @@ def callable(x: int | NAType) -> str | NAType:
     check(
         assert_type(s.map(callable, na_action=None), "pd.Series[str]"), pd.Series, str
     )
+    check(assert_type(s.map(callable), "pd.Series[str]"), pd.Series, str)
 
     series = pd.Series(["a", "b", "c"])
     check(assert_type(s.map(series, na_action=None), "pd.Series[str]"), pd.Series, str)
+    check(assert_type(s.map(series), "pd.Series[str]"), pd.Series, str)
+
+    s2: pd.Series[float] = pd.Series([1.0, pd.NA, 3.0])
+
+    def callable2(x: float) -> float:
+        return x + 1
+
+    check(
+        assert_type(s2.map(callable2, na_action="ignore"), "pd.Series[float]"),
+        pd.Series,
+        float,
+    )
+    check(
+        assert_type(s2.map(callable2), "pd.Series[float]"),
+        pd.Series,
+        float,
+    )
+    if TYPE_CHECKING_INVALID_USAGE:
+        s2.map(callable2, na_action=None)  # type: ignore[arg-type] # pyright: ignore[reportCallIssue, reportArgumentType]
+
+    s3: pd.Series[str] = pd.Series(["A", pd.NA, "C"])
+
+    def callable3(x: str) -> str:
+        return x.lower()
+
+    check(
+        assert_type(s3.map(callable3, na_action="ignore"), "pd.Series[str]"),
+        pd.Series,
+        str,
+    )
+    if TYPE_CHECKING_INVALID_USAGE:
+        s3.map(
+            callable3, na_action=None  # type: ignore[arg-type] # pyright: ignore[reportCallIssue, reportArgumentType]
+        )
+        s3.map(callable3)  # type: ignore[type-var] # pyright: ignore[reportCallIssue, reportArgumentType]
```

Comment:
I'm a little concerned that this won't pass the type checker.  A user should be able to do:
```python
s = pd.Series({"a", "b", "c"], dtype=str)
s.map(callable3)
``` 
That will execute correctly and should pass the type checker.

Reply:
Unfortunately i dont think the type system supports something like this at the moment.

All that `s.map(callable3)` sees is that the entries of `s` are string or none values and that none-values will be passed into `callable3` as they are. So it has to be able to handle them. The type system does not know if there actually are non-values in there or not, only that there might be. Most of the times the user probably wont know either if the data is loaded from an external file that comes from a sensor or a data repository.

If the user does know more than the type system he could either specify `na_action="ignore"`, which he should do anyway if the function does no handle the case explicitly, or add a type ignore to the line as one normally does when you have more information than can be contained in the type system (or understood by the type checkers).

Reply:
I understand your point of view.  But the issue here is that we need to support the "normal" usage of pandas.  The type system can't keep track of what is inside of a `Series` or `DataFrame`.  For example, if you have a `DataFrame` `df` with columns `"a"`, `"b"` then `df["a"]` is `Series[Any]` because we can't track the type inside the `DataFrame`.  

The default of the `na_action` argument in `Series.map()` is `None`, which means that people who know that their series has no non-null elements would now have to handle the case in the declared typing of `callable3()` to handle nulls, but that will likely break existing code.

We've had other cases where we were too restrictive in the stubs, and people complained, so we had to pull back. 

See https://github.com/pandas-dev/pandas-stubs/blob/main/docs/philosophy.md#narrow-vs-wide-arguments for a discussion of this issue.  IMHO, you're proposing that the stubs will be too narrow.

Reply:
Yeah, i think im just too much of a typing purist in this case. Because, just because your code doesnt cause runtime errors with the data that you are testing it with, doesnt mean that it isnt unsound from a typing perspective. Normally every interaction should be proveably valid just from the involved signatures. And i am using a type checker to actually ensure this.

But i guess that is not fully possible anyway with the setup of pandas.

Reply:
Yes, I agree.  We've chosen a middle ground with the stubs that supports the common ways people use pandas, but doesn't necessarily support all possible best typing practices.